server/storage/mvcc/kvstore_test.go

```diff
@@ -542,23 +542,19 @@ type hashKVResult struct {
 func TestHashKVWhenCompacting(t *testing.T) {
```

Comment:
The reason for the refactoring of this test case, is that the test used to run compactions/HashByRev  for 1000 rounds, using hash 9899 to 9999, with no guarantee that all compactions have finished before the end of the test.

Now, the test runs for as long as it takes to complete all compactions (within a limit of 10 seconds), using revisions from 9900 to 10000. This is beneficial because the problem from the initial issue, only appears after compacting revision 10000.